analyzer/src/main/groovy/com/novoda/sqlite/generator/SqliteAnalyzerPlugin.groovy

```diff
@@ -23,9 +24,26 @@ class SqliteAnalyzerPlugin implements Plugin<Project> {
     }
 
     private static void ensurePluginDependencies(Project project) {
-        def hasAndroid = project.plugins.hasPlugin('android') || project.plugins.hasPlugin('android-library')
-        if (!hasAndroid) {
-            throw new RuntimeException("The 'android' or 'android-library' plugin is required.")
+        if (!isApplication(project) && !isLibrary(project)) {
+            throw new ProjectConfigurationException("The 'android' or 'android-library' plugin is required.", null)
         }
     }
+
+    private static String variants(Project project) {
+        if (isApplication(project)) {
+            return 'applicationVariants'
+        } else if (isLibrary(project)) {
+            return 'libraryVariants'
+        }
+        throw new ProjectConfigurationException("The 'android' or 'android-library' plugin is required.", null)
```

Comment:
`nit:` the name of the plugins has been deprecated in favour of `com.android.application` and `com.android.library`. We should align logs with that.